### PR TITLE
completed hw3 01&03, need to create table for 02

### PR DESCRIPTION
**Pull request by imOnVacation**

### CODE REVIEW

File: hw3/03-forms.js [hunk 1]
```diff
@@ -1,10 +1,66 @@
...
+            res.writeHead(200, { 'Content-Type': 'text/html' })
+            res.end(postHTML)
+        } else if (req.url === '/submit') {
+            res.writeHead(200, { 'Content-Type': 'text/html' })
+            const params = new URLSearchParams(body)
+            res.write(`<h1>Name: ${params.get('name')}</h1>`)
+            res.write(`<h1>Email: ${params.get('email')}</h1>`)
+            if (!params.get('comment')) {
+                res.write(`<h1>Comment: n/a</h1>`)
+            } else res.write(`<h1>Name: ${params.get('name')}</h1>`)
```
Comment (vuong2-pdx) [inline hw3/03-forms.js:55]: Here if the user enters in a comment, the submit page is displaying the name section back to them when it should be displaying the comment section

File: hw3/02-url.js [hunk 2]
```diff
@@ -30,7 +36,18 @@ const server = http.createServer((req, res) => {
...
-
+    else {
+        const paramStr = url.search
+        if (!paramStr) {
+            res.write('No Query String in this URL')
+        } else {
+            res.writeHead(200, { 'Content-Type': 'text/html' })
+            const queryStr = new URLSearchParams(paramStr)
+            for (let key of queryStr.keys()) {
+                res.write(`<span>${key} ${queryStr.get(key)}</span><br/>`)
```
Comment (vuong2-pdx) [inline hw3/02-url.js:47]: for displaying your table - here you can append to an output string as you loop through the keys, then outside of the for loop you can do `res.write(outputString)`

**Review by vuong2-pdx — APPROVED** [review]
I left some comments, feel free to make changes as you see fit. Good job!

File: hw3/03-forms.js (hunk 1 above)
Comment (imOnVacation) [reply hw3/03-forms.js:55]: Oh right! Thanks for catching that mistake! :)

File: hw3/02-url.js (hunk 2 above)
Comment (imOnVacation) [reply hw3/02-url.js]: Thank you so much for your advice, I have updated the code accordingly.

File: hw3/03-forms.js [hunk 3]
```diff
@@ -1,10 +1,66 @@
...
+            const params = new URLSearchParams(body)
+            res.write(`<h1>Name: ${params.get('name')}</h1>`)
+            res.write(`<h1>Email: ${params.get('email')}</h1>`)
+            if (!params.get('comment')) {
+                res.write(`<h1>Comment: n/a</h1>`)
+            } else res.write(`<h1>Comment: ${params.get('comment')}</h1>`)
+            if (!params.get('newsletter')) {
+                res.write(`<h1>Newsletter: No, thank you.</h1>`)
+            } else res.write(`<h1>Newsletter: Yes, sign me up for the newsletter.</h1>`)
+        }
```
Comment (sashetov) [inline hw3/03-forms.js:59]: you may want to provide an error case when the URL and or method aren't matched, such as a 404, 405 error

File: hw3/02-url.js [hunk 4]
```diff
@@ -30,7 +36,21 @@ const server = http.createServer((req, res) => {
     }
 
     // Add your code here
-
+    else {
+        const paramStr = url.search
+        if (!paramStr) {
+            res.write('No Query String in this URL')
```
Comment (sashetov) [inline hw3/02-url.js:42]: it may make sense to have the status code be non-200 here, since its an error case

**Review by sashetov — APPROVED** [review]
i think its good to merge. i did offer some nitpicky suggestions that you can consider

**Pull request merged**